backend/hitas/views/housing_company.py

```diff
@@ -0,0 +1,245 @@
+        will validate the page number as well. if that raises an error then `PageNumberPagination` will raise `Http404`
+        which we don't want.
+        """
+
+        number = super(HitasPagination, self).get_page_number(request, paginator)
+
+        try:
+            paginator.validate_number(number)
+        except PageNotAnInteger:
+            raise InvalidPage()
+        except EmptyPage:
+            if int(number) < 1:
+                # If the given page number is an integer but not a positive one, then we throw the invalid page
+                # exception as well
+                raise InvalidPage()
+            else:
+                # if the given page is higher than maximum number of pages then let's just return the last page
+                number = paginator.num_pages
+
+        return number
+
+
+class HousingCompanyListSerializer(EnumSupportSerializerMixin, serializers.ModelSerializer):
+    id = serializers.SerializerMethodField()
+    name = serializers.CharField(source="display_name", max_length=1024)
+    state = serializers.SerializerMethodField()
+    address = serializers.SerializerMethodField()
+    area = serializers.SerializerMethodField()
+    date = serializers.SerializerMethodField()
+
+    def get_id(self, obj):
+        return obj.uuid.hex
+
+    def get_state(self, obj):
+        return obj.state.name.lower()
+
+    def get_address(self, obj):
+        return {
+            "street": obj.street_address,
+            "postal_code": obj.postal_code.value,
+            "city": obj.city,
+        }
+
+    def get_area(self, obj):
+        return {"name": obj.postal_code.description, "cost_area": obj.area}
+
+    def get_date(self, obj):
+        return obj.date
+
+    class Meta:
+        model = HousingCompany
+        fields = ["id", "name", "state", "address", "area", "date"]
+
+
+class HousingCompanyDetailSerializer(EnumSupportSerializerMixin, serializers.ModelSerializer):
+    id = serializers.SerializerMethodField()
+    name = serializers.SerializerMethodField()
+    state = serializers.SerializerMethodField()
+    address = serializers.SerializerMethodField()
+    area = serializers.SerializerMethodField()
+    date = serializers.SerializerMethodField()
+    real_estates = serializers.SerializerMethodField()
+
+    def get_name(self, obj):
+        return {
+            "display": obj.display_name,
+            "official": obj.official_name,
+        }
+
+    def get_id(self, obj):
+        return obj.uuid.hex
+
+    def get_state(self, obj):
+        return obj.state.name.lower()
+
+    def get_address(self, obj):
+        return {
+            "street": obj.street_address,
+            "postal_code": obj.postal_code.value,
+            "city": obj.city,
+        }
+
+    def get_area(self, obj):
+        return {"name": obj.postal_code.description, "cost_area": obj.area}
+
+    def get_date(self, obj):
+        return obj.date
+
+    def get_real_estates(self, obj):
```

Comment:
This could be made much simpler by creating serializer classes for Building and RealEstates

Reply:
agree, done